lucille-core/src/main/java/com/kmwllc/lucille/stage/ParseFilePath.java

```diff
@@ -0,0 +1,89 @@
+package com.kmwllc.lucille.stage;
+
+import com.kmwllc.lucille.core.ConfigUtils;
+import java.io.File;
+import java.util.Arrays;
+import java.util.Iterator;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.typesafe.config.Config;
+
+/**
+ * This stage will take a field that contains a file path (like  c:\directory\filename.txt) and parse out information about the
+ * path:
+ *   <br> - filename: The name of the file.
+ *   <br> - folder: The parent of the file.
+ *   <br> - path: The path to the file. (Normalized)
+ *   <br> - file_extension: The extension of the file.
+ *
+ * <br> <br> <b>Config Parameters:</b>
+ *
+ *   <br> - filePathField (String, Optional) - The field name that contains the file path. Defaults to "file_path".
+ *   <br> - fileSep (String, Optional) - The separator for your file system. Defaults to the operating system's separator.

```

Comment:
Should mention what the two supported options are.

Reply:
Also, what is the effect of setting fileSep? Is it needed for parsing the provided path? Or does it determine what separator will be used in the generated/normalized path? Or both?